test/TestGrains/MethodInterceptionGrain.cs

```diff
@@ -97,6 +128,62 @@ public Task<object> Invoke(MethodInfo methodInfo, InvokeMethodRequest request, I
         public Task<string> GetInputAsString(bool input) => Task.FromResult(input.ToString(CultureInfo.InvariantCulture));
 
         public Task<int> GetBestNumber() => Task.FromResult(38);
+    }
 
+#pragma warning disable 618
+    public class GrainCallFilterTestGrain : Grain, IGrainCallFilterTestGrain, IGrainCallFilter, IGrainInvokeInterceptor
+#pragma warning restore 618
+    {
+        private const string Key = GrainCallFilterTestConstants.Key;
+
+        // Note, this class misuses the context. It should not be stored for later use.
+        private IGrainCallContext context;
+
+        public async Task<string> CallWithBadInterceptors(bool early, bool mid, bool late)
+        {
+            if (late)
+            {
+                this.context.Arguments[2] = false;
+                await this.context.Invoke();
+            }
+
+            return $"I will {(early ? string.Empty : "not ")}misbehave!";
+        }
+
+        public Task<string> GetRequestContext() => Task.FromResult((string)RequestContext.Get(Key) + "6");
+
+        public async Task Invoke(IGrainCallContext ctx)
+        {
+            //
+            // NOTE: this grain demonstrates incorrect usage of grain call interceptors and should not be used
+            // as an example of proper usage. Specifically, storing the context for later execution is invalid.
+            //
+
+            this.context = ctx;
+            var value = RequestContext.Get(Key) as string;
+            if (value != null) RequestContext.Set(Key, value + '4');
+            await ctx.Invoke();
+            this.context = null;
+        }
+
+        public async Task<object> Invoke(MethodInfo method, InvokeMethodRequest request, IGrainMethodInvoker invoker)
```

Comment:
can you implement the interfaces explicitly? It's still a little confusing without going back and forth to know which are these methods for, especially since these test grains implement both the new and the deprecated way of intercepting.
Also, it's probably makes sense logically too. These are cross-cutting concerns not related to the domain, so having them implemented explicitly instead of publicly makes sense as a good example. Not that the behavior will be any different though